frontend/src/components/App/Settings/SettingsCluster.tsx

```diff
@@ -37,19 +38,53 @@ function isValidClusterNameFormat(name: string) {
   return regex.test(name);
 }
 
+interface ClusterSelectorProps {
+  currentCluster?: string;
+  clusters: string[];
+}
+
+function ClusterSelector(props: ClusterSelectorProps) {
+  const { currentCluster = '', clusters } = props;
+  const history = useHistory();
+  const { t } = useTranslation('glossary');
+
+  return (
+    <FormControl variant="outlined" margin="normal" sx={{ minWidth: 250 }}>
+      <Select
```

Comment:
A nice to have: This could take an "error={props.error}" attribute to show when it's not valid. Like when the name isn't there.

Reply:
I feel this would clash with the error message, since the selector would turn red even though it only displays valid clusters. would be less confusing IMO to leave this out but if you don't think so I can add this in